src/main/index.js

```diff
@@ -60,8 +60,14 @@ export { downloadBlob } from './utils';
  * @property {number} progress Job progress, from 0 to 1.
  * @property {'pending' | 'ongoing' | 'finished'} status Job status.
  * @property {Blob} [imageBlob] Finished image blob.
+ * @property {Array<SourceLoadError>} [sourceLoadErrors] Array of sources that encountered at least one 'tileloaderror' or 'imageloaderror'.
  */
 
+/**
+ * @typedef {Object} SourceLoadError
+ * @property {ol.source} source ol source that encountered at least one 'tileloaderror' or 'imageloaderror'.
```

Comment:
Not sure it's going to be explicit enough. I like the tile path and error status combo. Did you inspect `event.error` property ? Nothing interesting in there ?

Reply:
You are right the error info is not ideal. However, the `TileSourceEvent {type: "tileloaderror", target: XYZ, tile: ImageTile}` does not have an error property that gives more error info like its code, just the type "tileloaderror". I don't see how to get the error code without touching on `setTileLoadFunction`.
The current source property allows to inspect which urls returned errors. I could also add a logic that extracts the url instead of passing the whole source object, but this logic will have to target different properties depending on the source.

Reply:
I think giving the url is a good way the user can check what's going on, for a start.
Implement an extraction function that could test the type of the source object to pick the correct strategy.

Reply:
Ok I adapted accordingly, `getUrl(s)` actually does the job